nevergrad/functions/unit_commitment/core.py

```diff
@@ -0,0 +1,58 @@
+import numpy as np
+import nevergrad as ng
+from ..base import ExperimentFunction
+
+class UnitCommitmentProblem1(ExperimentFunction):
+    """Model that uses conventional implementation for emi-continuous variables
+    The model is adopted from Pyomo model 1: Conventional implementation for emi-continuous variables
+    (https://jckantor.github.io/ND-Pyomo-Cookbook/04.06-Unit-Commitment.html)
+    The constraints are added to the objective with a heavy penalty for violation.
+
+    Parameters
+    ----------
+    T_points: int
+        number of time points
+    N_generators: int
+        number of generators
+    penalty_weight: float
+        weight to penalize for violation of constraints
+    """
+    def __init__(self, T_points: int = 13,
+                 N_generators: int = 3,
+                 penalty_weight : float = 10000
+                 ) -> None:
+        params = {x: y for x, y in locals().items() if x not in ["self", "__class__"]}  # for copying
+        # Demand for certain time period
+        self.T_points = T_points
+        self.demands = np.array([100 + 100*np.random.uniform() for t in range(self.T_points)])
```

Comment:
please direcly use `np.random.uniform` parameters, it can take a lower bound, an upper bound and a size.
You may also consider using a possibly seeded random generator for reproducibility, but that really depends how you want your function to behave. The benchmarks should be reproducible without it anyway since the standard numpy random number generator is seeded before running